v2/r/sample.R

```diff
@@ -0,0 +1,101 @@
+# Yelp API v2.0 code sample.
+# 
+# This program demonstrates the capability of the Yelp API version 2.0
+# by using the Search API to query for businesses by a search term and location.
+# 
+# Please refer to http://www.yelp.com/developers/documentation for the API documentation.
+# 
+# This program requires some R libraries including "httr", which you can install via:
+# `packages.install("httr")`
+# `packages.install("httpuv")`
+# etc.
+# 
+# Sample usage of the program:
+# `> source("sample.R")`
+# (output to the screen)
+# or
+# `R CMD BATCH sample.R`
+# (output to sample.Rout)
+
+# Required packages
+require(httr)
+require(httpuv)
+require(jsonlite)
+require(base64enc)
+
+# Your credentials, from https://www.yelp.com/developers/manage_api_keys
+consumerKey = ""
+consumerSecret = ""
+token = ""
+token_secret = ""
+
+yelp_query <- function(path, query_args) {
+  # Use OAuth to authorize your request.
+  myapp <- oauth_app("YELP", key=consumerKey, secret=consumerSecret)
+  sig <- sign_oauth1.0(myapp, token=token, token_secret=token_secret)
+
+  # Build Yelp API URL.
+  scheme <- "https"
+  host <- "api.yelp.com"
+  yelpurl <- paste0(scheme, "://", host, path)
+
+  # Make request.
+  results <- GET(yelpurl, sig, query=query_args)
+
+  # If status is not success, print some debugging output.
+  HTTP_SUCCESS <- 200
+  if (results$status != HTTP_SUCCESS) {
+      print(results)
+  }
+  return(results)
+}
+
+yelp_search <- function(term="dinner", location="Boston, MA", limit=3) {
```

Comment:
1. Defaults for search but not business: Hm, good point.

Rather than adding a default business, I'll remove the Search defaults for term and location.

Having a default for limit seems more reasonable; I'll fiddle with that so that the `yelp_search()` function has a high limit for the default case, but `demo()` calls it with a low limit for easier browsing.

2. As with the Python example in the same repo: This is an example of a common flow demonstrating how to build, make, and parse a request --- it's not a full clientlib.  I don't plan on covering the Phone Search endpoint or all the various options for Search and Business.